src/test/java/org/sagebionetworks/bridge/services/AccountServiceTest.java

```diff
@@ -1,167 +1,954 @@
+        assertEquals(account.getModifiedOn().toString(), MOCK_DATETIME.withZone(UTC).toString());
+        assertEquals(account.getStatus(), ENABLED);
+        assertEquals(account.getPhoneVerified(), TRUE);
+        verify(mockAccountDao).updateAccount(account, null);
+    }
+
+    @Test
+    public void verifyPhoneUsingAccount() {
+        Account account = Account.create();
+        account.setStudyId(TEST_STUDY_IDENTIFIER);
+        account.setId(USER_ID);
+        account.setStatus(UNVERIFIED);
+        account.setPhoneVerified(FALSE);
+
+        service.verifyChannel(AuthenticationService.ChannelType.PHONE, account);
+
+        assertEquals(account.getStatus(), ENABLED);
+        assertEquals(account.getPhoneVerified(), TRUE);
+        // modifiedOn is stored as a long, which loses the time zone of the original time stamp.
+        assertEquals(account.getModifiedOn().toString(), MOCK_DATETIME.withZone(UTC).toString());
+        assertEquals(account.getStatus(), ENABLED);
+        assertEquals(account.getPhoneVerified(), TRUE);
+        verify(mockAccountDao).updateAccount(account, null);
+    }
+
+    @Test
+    public void verifyPhoneUsingAccountNoChangeNecessary() {
+        Account account = Account.create();
+        account.setId(USER_ID);
+        account.setStatus(ENABLED);
+        account.setPhoneVerified(TRUE);
+
+        service.verifyChannel(AuthenticationService.ChannelType.PHONE, account);
+        verify(mockAccountDao, never()).updateAccount(any(), any());
+    }
+
+    @Test
+    public void verifyPhoneWithDisabledAccountMakesNoChanges() {
+        Account account = Account.create();
+        account.setStatus(DISABLED);
+
+        service.verifyChannel(AuthenticationService.ChannelType.PHONE, account);
+        verify(mockAccountDao, never()).updateAccount(any(), eq(null));
+        assertEquals(account.getStatus(), DISABLED);
+    }
+
+    @Test
+    public void changePasswordSuccess() throws Exception {
+        // Set up test account
+        Account account = Account.create();
+        account.setStudyId(TEST_STUDY_IDENTIFIER);
+        account.setId(USER_ID);
+
+        // execute and verify
+        service.changePassword(account, ChannelType.EMAIL, DUMMY_PASSWORD);
+        verify(mockAccountDao).updateAccount(accountCaptor.capture(), eq(null));
+
+        Account updatedAccount = accountCaptor.getValue();
+        assertEquals(updatedAccount.getId(), USER_ID);
+        assertEquals(updatedAccount.getModifiedOn().getMillis(), MOCK_DATETIME.getMillis());
+        assertEquals(updatedAccount.getPasswordAlgorithm(), PasswordAlgorithm.DEFAULT_PASSWORD_ALGORITHM);
+        assertEquals(updatedAccount.getPasswordModifiedOn().getMillis(), MOCK_DATETIME.getMillis());
+        assertTrue(updatedAccount.getEmailVerified());
+        assertNull(updatedAccount.getPhoneVerified());
+        assertEquals(updatedAccount.getStatus(), ENABLED);
```

Comment:
verifyStatus behaves that way, but changePassword always sets the account to enabled if it's going to change the password. I added a value to the status field in these tests (rather than null) to verify this doesn't stop the account from being enabled.